op-batcher/batcher/driver.go

```diff
@@ -289,8 +290,11 @@ func (l *BatchSubmitter) loop() {
 	defer ticker.Stop()
 
 	publishAndWait := func() {
-		l.publishStateToL1(queue, receiptsCh)
+		l.publishStateToL1(queue, receiptsCh, &daWaitGroup)
```

Comment:
In this new architecture, the `mainLoop` talks to the `daQueue` which then talks to `queue`.

With that in mind. I think if we have a `DaPublisher` (similar to the `TxManager`) we can abstract away all details of the `TxManager`... all in all, the `TxManager` is an impl detail of the `DaPublisher`.

From the `BatchSubmitter` perspective, It doesn't need to know about the `TxManager`. It only needs to be able to "publish txData", and for each txData to be published it needs a "receipt"... a resulst to know wether publishing was successful or a failure.

Also, better to have logic from the "publisher" in it's own class and outside the BatchSubmitter, so moving `sendTransaction` out. That would give us more control to what "state" is accesible, and we don't. have 2 goroutines accesing state in concurrent manner for thing that don't support that